datastream-kafka-connector/src/test/java/com/linkedin/datastream/connectors/kafka/TestAbstractKafkaConnector.java

```diff
@@ -50,6 +62,16 @@ public void testConnectorRestartCalled() {
         Duration.ofSeconds(10).toMillis());
     Assert.assertTrue(connector.getCreateTaskCalled() >= 3);
 
+    // Verify metric for nanny task restarts get incremented
+    Gauge<Long> metric = DynamicMetricsManager.getInstance()
+        .getMetric("test" + "." + TestKafkaConnector.class.getSimpleName() + "." + "numTaskRestarts");
+    Assert.assertNotNull(metric);
+    Assert.assertEquals(metric.getValue(), Long.valueOf(1));
+
+    // Verify that metrics created through DynamicMetricsManager match those returned by getMetricInfos() given the
+    // connector name of interest.
+    MetricsTestUtils.verifyMetrics(connector, DynamicMetricsManager.getInstance());
```

Comment:
By default, this verification won't include metrics that do not start with the class name of the `metricsAware` object in question, i.e. it won't consider the metric `test.TestKafkaConnector.numTaskRestarts` because the it doesn't start with `TestKafkaConnector`. To have it include that metric, you'll need to pass in a predicate that explicitly filters such metrics, e.g. `s -> s.startsWith("test")`